### ISSUE TITLE
hw3 review

### ISSUE THREAD

**Issue by vladimirdubal** [mkcomer237/DAT_SF_16#5]
Hi Max, 

I was supposed to review Tanguy's homework, but it does not look like he submitted any, 
so Francesco asked me to review someone else's, and I chose yours. 

I am a newbie, and overall it was helpful to read your code, as that helped me see
many things that I missed. So my comments are mostly around that. 

Here you replace all 0 values with nan. 

for column in missing_features:  
    df[column].replace(0, np.nan, inplace=True)

Then later you replace it by sampling some random value that lies around the mean and the std, at least  
that is my understanding of the following code: 
np.random.normal(feature_mean,feature_std)

That is good! 
I failed to do that in my code. I.e. I did not address the 0 values. 
(also good assumption that you did not include pregnant in missing features!)

Here you are normalizing the data: 
 df[feature] = (df[feature] - feature_mean)/feature_std

 I failed to do that as well, as I was struggling with Python in general. 

 Maybe of interest that even without normalizing or replacing the 0 values, 
 I got a cross-validation score for knn of 0.743, whereas yours was 0.75.   

With regards to Naive-Bayes, I had used
 from sklearn.naive_bayes import MultinomialNB
instead of 

from sklearn.naive_bayes import GaussianNB

Not sure what the difference is in using either. Also, on my end there was some
cludgyness in how I used the code (due to a lack of clear understanding of what 
I was doing), and I also did not use cvv for Bayes. 

Overall, your code is clear/clean and it helped me further understand a 
few points I had completely missed. 

Apologies if I missed any errors you might have made or do not have much to add in 
terms of suggesting improvements. The only thing for me, personally, is that I would find a few additional comments helpful, but that is mostly because I am not yet well versed at reading code, and would could argue that the code should be able to speak for itself.  

Vladimir

@ghego @breucopter

**Comment by ghego** [comment]
Hey Max,
good job with HW3!

a couple of comments:
- good intuition on the 0s !
- why not using the CV function from scikit learn instead of the one from the lab?
- Did you understand the difference between GaussianNB and MultinomialNB?

Keep rocking!

**Comment by mkcomer237** [comment]
Thanks for the comments @vladimirdubal!

Francesco, I couldn't see how to easily do the cross validation in sklearn and I was running out of time, so I just used our function from class.  Someone else showed me since then so I think i have it down now.  I'm not sure I understand the difference between GuassianNB and MultinomialNB - from the documentation Gaussian seemed more standard so that's what I used.  

Max

**Comment by ghego** [comment]
Have a look here for CV: http://scikit-learn.org/stable/modules/cross_validation.html

and here for the various NB options: http://scikit-learn.org/stable/modules/naive_bayes.html

The difference is the assumption on the distribution of the data and if data is continuous or discrete.
